### PR TITLE
Added changelog entries for the upcoming 0.16.1 release

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
This PR modifies `CHANGELOG.md` file to document changes from 0.16.0 tag to future 0.16.1 tag.

The writeup assumes that gh-1612 is merged, so this should only be merged after gh-1612 is in.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,25 @@ All notable changes to this project will be documented in this file.
...
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.16.1] - Apr. XX, 2024
+
+This is a bug-fix release, which also provides a change needed by ``numba_dpex`` project to support dispatching kernels
+consuming instances of ``sycl::kernel_accessor`` template type.
+
+### Changed
+
+* Changed behavior of ``dpctl.tensor.usm_ndarray.__dlpack_device__`` method to return device id of the parent unpartition device if array is allocated on a sub-device instead of raising an exception: [#1604](https://github.com/IntelPython/dpctl/pull/1604)
```
Comment (diptorupd) [inline CHANGELOG.md:14]: s `unpartition` a SYCL term? If not, should be "unpartitioned"

**Comment by ndgrigorian** [comment]
@oleksandr-pavlyk 
Fixed section should also mention changes which resolved #1570 from #1608

**Review by ndgrigorian — APPROVED** [review]

**Comment by oleksandr-pavlyk** [comment]
@ndgrigorian I would need to push one more change here to account for fix in gh-1615

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,27 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.16.1] - Apr. XX, 2024
+
+This is a bug-fix release, which also provides a change needed by ``numba_dpex`` project to support dispatching kernels
+consuming instances of ``sycl::kernel_accessor`` template type.
```
Comment (diptorupd) [inline CHANGELOG.md:10]: s/``sycl::kernel_accessor``/``sycl::local_accessor``

**Pull request merged**